src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1369,9 +1369,18 @@ private static bool IsCompilerAnalyzer(DiagnosticAnalyzer analyzer)
 
         public void Dispose()
         {
-            this.CompilationEventQueue?.TryComplete();
-            this.DiagnosticQueue?.TryComplete();
-            _queueRegistration.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
```

Comment:
Since the enclosing type implements `IDisposable` and is not sealed, I fixed the implementation to adhere to requirements of the `IDisposable` interface. I believe this change can be moved to a separate PR if needed.